`sockets` fixture is broken on PyPy3.x

The `sockets` fixture throws an error during teardown, when running the test suite on PyPy3.10 and PyPy3.11, both version 7.3.19. My guess is that it's trying to close a socket that's already been closed. I'm cutting the test output below on the first error, since they're all the same.

We need PyPy support in this package since ipykernel started depending on it. pyzmq itself passes all tests on PyPy, so I think the bug is in zmq-anyio.

```pytb
$ uv venv -p pypy3.11
Using PyPy 3.11.11 interpreter at: /usr/bin/pypy3.11
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
$ uv pip install -e .[test]
Resolved 21 packages in 8ms
░░░░░░░░░░░░░░░░░░░░ [0/21] Installing wheels...                                                                                       warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 21 packages in 26ms
 + anyio==4.8.0
 + anyioutils==0.7.1
 + attrs==25.1.0
 + cffi==1.17.1
 + coverage==7.6.12
 + idna==3.10
 + iniconfig==2.0.0
 + mypy==1.15.0
 + mypy-extensions==1.0.0
 + outcome==1.3.0.post0
 + packaging==24.2
 + pluggy==1.5.0
 + pycparser==2.22
 + pytest==8.3.5
 + pyzmq==26.2.1
 + ruff==0.9.10
 + sniffio==1.3.1
 + sortedcontainers==2.4.0
 + trio==0.27.0
 + typing-extensions==4.12.2
 + zmq-anyio==0.3.6 (from file:///tmp/zmq-anyio)
$ uv run pytest -vvx
========================================================= test session starts =========================================================
platform linux -- Python 3.11.11[pypy-7.3.19-final], pytest-8.3.5, pluggy-1.5.0 -- /tmp/zmq-anyio/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /tmp/zmq-anyio
configfile: pyproject.toml
plugins: anyio-4.8.0
collected 46 items                                                                                                                    

tests/test_socket.py::test_context[asyncio] PASSED                                                                              [  2%]
tests/test_socket.py::test_arecv_multipart[asyncio] PASSED                                                                      [  4%]
tests/test_socket.py::test_arecv_multipart[asyncio] ERROR                                                                       [  4%]

=============================================================== ERRORS ================================================================
_________________________________________ ERROR at teardown of test_arecv_multipart[asyncio] __________________________________________

contexts = {<zmq.Context(1 socket) at 0x7f934d88a800 closed>}

    @pytest.fixture
    def sockets(contexts):
        sockets = []
        yield sockets
        # ensure any tracked sockets get their contexts cleaned up
        for socket in sockets:
            contexts.add(socket.context)
    
        # close sockets
        for socket in sockets:
>           socket.close(linger=0)

tests/conftest.py:79: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/pypy3.11/site-packages/zmq/sugar/socket.py:261: in close
    super().close(linger=linger)
.venv/lib/pypy3.11/site-packages/zmq/backend/cffi/socket.py:157: in close
    self.set(zmq.LINGER, linger)
.venv/lib/pypy3.11/site-packages/zmq/backend/cffi/socket.py:228: in set
    _retry_sys_call(
.venv/lib/pypy3.11/site-packages/zmq/backend/cffi/utils.py:71: in _retry_sys_call
    _check_rc(rc)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

rc = -1, errno = 88, error_without_errno = True

    def _check_rc(rc, errno=None, error_without_errno=True):
        """internal utility for checking zmq return condition
    
        and raising the appropriate Exception class
        """
        if rc == -1:
            if errno is None:
                from zmq.backend import zmq_errno
    
                errno = zmq_errno()
            if errno == 0 and not error_without_errno:
                return
            from zmq import EAGAIN, ETERM
    
            if errno == EINTR:
                raise InterruptedSystemCall(errno)
            elif errno == EAGAIN:
                raise Again(errno)
            elif errno == ETERM:
                raise ContextTerminated(errno)
            else:
>               raise ZMQError(errno)
E               zmq.error.ZMQError: Socket operation on non-socket

.venv/lib/pypy3.11/site-packages/zmq/error.py:151: ZMQError
======================================================= short test summary info =======================================================
ERROR tests/test_socket.py::test_arecv_multipart[asyncio] - zmq.error.ZMQError: Socket operation on non-socket
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================================== 2 passed, 1 error in 0.62s ======================================================
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -14,7 +14,15 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: [ '3.9', '3.10', '3.11', '3.12', '3.13' ]
+        python-version:
+          - '3.9'
+          - '3.10'
+          - '3.11'
+          - '3.12'
+          - '3.13'
+          - 'pypy-3.9'
+          - 'pypy-3.10'
+          - 'pypy-3.11'
 
     steps:
       - name: Checkout repository

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,6 +1,5 @@
 import os
 import signal
-import socket
 from threading import Thread
 
 import zmq
@@ -76,7 +75,8 @@ def sockets(contexts):
 
     # close sockets
     for socket in sockets:
-        socket.close(linger=0)
+        if not socket.closed:
+            socket.close(linger=0)
 
 
 @pytest.fixture

